test/arrayops.jl

```diff
@@ -1705,6 +1705,30 @@ end
     @test istriu([1 2 0; 0 4 1])
 end
 
+#issue 49021
+@testset "reverse cartesian indices" begin
+    @test isequal(reverse(CartesianIndices((2, 3))), CartesianIndices((2:-1:1, 3:-1:1)))
+    @test isequal(reverse(CartesianIndices((2:5, 3:7))), CartesianIndices((5:-1:2, 7:-1:3)))
+    @test isequal(reverse(CartesianIndices((5:-1:2, 7:-1:3))), CartesianIndices((2:1:5, 3:1:7)))
+end
+
+@testset "reverse cartesian indices dim" begin
+    A = CartesianIndices((2, 3, 5:-1:1))
+    @test isequal(reverse(A, dim=1), CartesianIndices((2:-1:1, 3, 5:-1:1)))
+    @test isequal(reverse(A, dim=3), CartesianIndices((2, 3, 1:1:5)))
```

Comment:
Same here.
BTW, it would be good to add some stability test to avoid possible regression on const propagation
```julia
julia> A = CartesianIndices((2, 3, 5:-1:1))
CartesianIndices((2, 3, 5:-1:1))

julia> f13(x) = reverse(x; dims = (1, 3))
f13 (generic function with 1 method)

julia> f1(x) = reverse(x; dims = 1)
f1 (generic function with 1 method)

julia> @test @inferred(f1(A)) === CartesianIndices((2:-1:1, 3, 5:-1:1))
Test Passed

julia> @test @inferred(f13(A)) === CartesianIndices((2:-1:1, 3, 1:1:5))
Test Passed
```